### DISCUSSION TITLE
Is it possible to decode a nullopt as null for optional fields in a struct?

### DISCUSSION THREAD

**Discussion by recsater** [danielaparker/jsoncons#497] [Q&A] [answered]
Example

```c++
struct A {
  optional<int> a;
}
JSONCONS_N_MEMBER_TRAITS(A, 0, a)

int main() {
  std::string result;
  jsoncons::encode_json(A{}, result); // result -> "{}"
}
```

If result were `{ "a": null }` with some option (like ignore_null: default true), it would have been better.

**Comment by danielaparker** [comment, marked as answer]
Either 

    JSONCONS_ALL_MEMBER_TRAITS(A, a) 

or

    JSONCONS_N_MEMBER_TRAITS(A, 1, a)

would produce

    { "a": null }

But non-mandatory means that the member can be omitted.
